default.nix

```diff
@@ -1,17 +1,64 @@
-with import (builtins.fetchGit {
-  url = https://github.com/NixOS/nixpkgs-channels;
-  ref = "nixos-18.09-small";
-  rev = "95fed28ac372c61eb83c87ad97c24b0f957827bf";
-}) {};
+let
+  # Commit of the Nixpkgs repository that we want to use.
+  nixpkgsVersion = {
+      date = "2020-10-04";
+      rev = "d191ee22f0b0c91070f9f4c79210d4e398df0b93";
+      tarballHash = "13fb4cgxxc47shmzbm0rawfxjnilf7r4zq1vwc0z6s3lxijl9wl9";
+    };
+
+  # Nix files that describe the Nixpkgs repository. We evaluate the expression
+  # using `import` below.
+  pkgs = import (fetchTarball {
+      url = "https://github.com/nixos/nixpkgs/archive/${nixpkgsVersion.rev}.tar.gz";
+      sha256 = nixpkgsVersion.tarballHash;
+    }) {};
+
+  python = pkgs.python36.withPackages (ps: [ps.sphinx ps.sphinx_rtd_theme ps.livereload]);
+
+  build =
+    pkgs.writeShellScriptBin "postgrest-docs-build"
+      ''
+        set -euo pipefail
+
+        # clean previous build, otherwise some errors might be supressed
+        rm -rf _build
+
+        ${python}/bin/sphinx-build -W -b html -a -n . _build
+      '';
+
+  serve =
+    pkgs.writeShellScriptBin "postgrest-docs-serve"
+      ''
+        set -euo pipefail
+
+        # livereload_docs.py needs to find "sphinx-build"
+        PATH=${python}/bin:$PATH
+
+        ${python}/bin/python livereload_docs.py
+      '';
+
+  spellcheck =
+    pkgs.writeShellScriptBin "postgrest-docs-spellcheck"
+      ''
+        set -euo pipefail
+
+        cat *.rst                                                                              \
+         | grep -v '^\(\.\.\|  \)'                                                             \
+         | sed 's/`.*`//g'                                                                     \
+         | ${pkgs.aspell}/bin/aspell -d ${pkgs.aspellDicts.en}/lib/aspell/en_US -p ./postgrest.dict list \
+         | tee misspellings
+        test ! -s misspellings
+      '';
+in
+
+pkgs.stdenv.mkDerivation {
```

Comment:
Nitpick: We could use pkgs.buildEnv here, making it clearer that we just want to put together an environment, e.g.

```
buildEnv {
  name = "postgrest-docs";
  paths = [ build serve spellcheck ];
}
```

Reply:
Hm. If I do that, then I can't use `inherit` anymore and the tools are not available via `nix-env -iA build` etc.